src/connect/components/tree-view/tree-view.stories.tsx

```diff
@@ -108,119 +50,108 @@ const TreeViewImpl = (args: ConnectTreeViewProps) => {
         onDragEnd,
         ...treeViewProps
     } = args;
-    const [items, setItems] = useState(argItems);
     const [disableHighlight, setDisableHighlight] = useState(false);
+    const { setItems } = useItemsContext();
 
     const onItemClickHandler: ConnectTreeViewProps['onItemClick'] = (
         e,
         item,
     ) => {
         onItemClick?.(e, item);
-        setItems(prevState => {
-            const newTree = traverseTree(prevState, treeItem => {
-                if (treeItem.id === item.id) {
-                    treeItem.isSelected = true;
-                    treeItem.expanded = !treeItem.expanded;
-                } else {
-                    treeItem.isSelected = false;
-                }
-
-                return treeItem;
-            });
+        setItems((itemsContext) => itemsContext.map((itemContext) => {
```

Comment:
in a next commit I'll add specific hooks and utils to perform this actions in the UI state (open folder, select folder, delete UI info, create virtual node, etc.)